mlflow/models/utils.py

```diff
@@ -34,6 +34,26 @@
 except ImportError:
     HAS_SCIPY = False
 
+try:
+    from pyspark.sql import DataFrame as SparkDataFrame
+    from pyspark.sql import Row
+    from pyspark.sql.types import (
+        ArrayType,
+        BinaryType,
+        DateType,
+        FloatType,
+        IntegerType,
+        ShortType,
+        StructType,
+        TimestampType,
+    )
```

Comment:
Let's import those inside corresponding functions, not at top level

Reply:
multiple functions use these types. May not be optimal to move these into the corresponding functions.